src/deluge/gui/views/instrument_clip_view.cpp

```diff
@@ -2148,33 +2148,45 @@ void InstrumentClipView::adjustProbability(int32_t offset) {
 								prevBase = false;
 							}
 							else {
-								// See if there's a prev-base
-								if (probabilityValue > 0 && probabilityValue < kNumProbabilityValues
-								    && getCurrentInstrumentClip()->doesProbabilityExist(
-								        editPadPresses[i].intendedPos, probabilityValue,
-								        kNumProbabilityValues - probabilityValue)) {
+								if (probabilityValue == 0) {
```

Comment:
It can't follow the same logic as prevBase because the prevBase logic searches the row for any previous notes with the same probability. The NOT FILL condition doesn't work this way, it must be independent from other FILL notes. You could have just NOT FILL notes in a clip if you wanted, that's why the probability 0 is handled differently. The logic here is to allow to scroll in this order (without checking other notes):  0 ==> 0+prevBase ==> 1.

I will do the change to if (probability == 0 || ...) that you suggested!